ci/tasks/scripts/lint.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e -u -o pipefail
+
+./src/ci/tasks/scripts/with-docker.sh
+
+workspace_dir="${PWD}"
+cd src
+
+make lint
+
+cd "${workspace_dir}"
```

Comment:
Pushd and popd could work nicely for this?

Reply:
it adds additional overhead for reading the script.
If anything goes wrong, it goes wrong in unusual ways (usually due to the stack not being popped).

Let's just keep it simple